src/createAppContainer.js

```diff
@@ -227,7 +239,7 @@ export default function createNavigationContainer(Component) {
         startupState = Component.router.getStateForAction(action);
       }
 
-      // Pull persisted state from AsyncStorage
+      // Pull user-provided persisted state
       if (startupStateJSON) {
         try {
           startupState = JSON.parse(startupStateJSON);
```

Comment:
currently `loadNavigationState` is expected to return a json string and we call JSON.parse on that string here to get an object.

I'd vote for `loadNavigationState` returning an object instead of a string (It'd just require moving this JSON.parse into the user-provided function).

From a developer perspective, for whatever reason, if I want to manipulate the state, I'd prefer to work with an object and return that object, rather than converting a json string (from asyncStorage) to an object, then manipulate what I need and then stringifying that, passing it to react-navigation which would call JSON.parse again. wdyt @brentvatne ?

Reply:
I went ahead and did that in https://github.com/react-navigation/react-navigation-native/pull/21/commits/0d7ad40e868bb8ce8a3c00613d03f8c873360798

Reply:
that seems totally reasonable @vonovak! apologies for the slow reply